### PR TITLE
Extend macOS support

### PR DESCRIPTION
**Pull request by koecher**
Updates candi in several places to get better macOS support.

* Works on a fresh and current macbook pro (2017 / Sierra / 10.12) with homebrew openmpi against Xcode-clang compiler.
* Tested packages: parmetis, superlu_dist, hdf5, p4est, trilinos (auto/12), dealii (v8.5.0)
* Untested packages: arpack-ng, opencascade, petsc, slepc

The platform installation notes have to be added (e.g. how to install homebrew, openmpi) and the README.md should be updated

### CODE REVIEW

File: candi.sh [hunk 1]
```diff
@@ -578,11 +575,8 @@ guess_platform() {
     elif [ -x /usr/bin/sw_vers ]; then
         local MACOSVER=$(sw_vers -productVersion)
         case ${MACOSVER} in
-            10.4*)    echo tiger;;
-            10.5*)    echo leopard;;
-            10.6*)    echo snowleopard;;
-            10.7*)    echo lion;;
-            10.8*)    echo mountainlion;;
+		    10.11*)   echo macosx;;
```
Comment (koecher) [inline candi.sh]: TODO: FIXME: here is the intendation broken

File: candi.sh [hunk 2]
```diff
@@ -605,6 +599,40 @@ guess_platform() {
     fi
 }
 
+guess_ostype() {
+    # Try to guess the operating system type (ostype)
+    if [ -f /usr/bin/cygwin1.dll ]; then
+        echo cygwin
+
+    elif [ -f /etc/fedora-release ]; then
+	    echo linux
```
Comment (koecher) [inline candi.sh]: TODO: FIXME: intendation

File: candi.sh (hunk 1 above)
Comment (koecher) [inline candi.sh]: TODO: FIXME: insert here elcapitan instead of macosx

**Comment by koecher** [comment]
Sorry, the indentation is broken in several places due to non-experience to xcode. I'll update this.
Would be nice if anyone test this branch on a linux machine and also on macOS 10.12 or 10.11.

File: candi.sh [hunk 3]
```diff
@@ -605,6 +599,40 @@ guess_platform() {
...
+    
+    elif [ -x /usr/bin/sw_vers ]; then
+        echo macos
+    
+    elif [ -x /usr/bin/lsb_release ]; then
+        local DISTRO=$(lsb_release -i -s)
+        local CODENAME=$(lsb_release -c -s)
+        local DESCRIPTION=$(lsb_release -d -s)
+        case ${DISTRO}:${CODENAME}:${DESCRIPTION} in
+            *:*:*Ubuntu*\ 12*)     echo linux;;
```
Comment (tjhei) [inline candi.sh]: Why are you adding this detection using lsb_release? It is not guaranteed to be available, see https://www.unixmen.com/linux-troubleshooting-fix-lsb_release-command-found-centos/ for example

File: deal.II-toolchain/packages/trilinos.package [hunk 4]
```diff
@@ -105,24 +105,37 @@ CONFOPTS="${CONFOPTS} \
...
+	    exit 1
+    fi
+
     CONFOPTS="\
         ${CONFOPTS} \
         -D TPL_ENABLE_ParMETIS:BOOL=ON \
-        -D TPL_ParMETIS_LIBRARIES:FILEPATH=${PARMETIS_DIR}/lib/libparmetis.so \
+        -D TPL_ParMETIS_LIBRARIES:FILEPATH=${PARMETIS_DIR}/lib/libparmetis.${LDSUFFIX} \
         -D TPL_ParMETIS_INCLUDE_DIRS:PATH=${PARMETIS_DIR}/include"
+###### TODO?  -D HAVE_PARMETIS_VERSION_4_0_3=ON"
```
Comment (tjhei) [inline deal.II-toolchain/packages/trilinos.package]: what is this todo about?

File: deal.II-toolchain/platforms/supported/elcapitan.platform [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# macOS 10.11 (El Capitan)
+#
+# Mac support is experimental and requires:
```
Comment (tjhei) [inline deal.II-toolchain/platforms/supported/elcapitan.platform]: can you remove the macosx.platform file if it is no longer needed?

Comment (koecher) [inline deal.II-toolchain/platforms/supported/elcapitan.platform]: coming with the next update, where I'll write the platform notes

File: candi.sh (hunk 3 above)
Comment (koecher) [inline candi.sh]: It is the last option for candi, centos/rhel are found earlier.
I simply copied the guess_platform function and simplified it in this place here.

File: deal.II-toolchain/platforms/supported/sierra.platform [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+# and then run
+# $ xcodebuild -license
+# to accept the license.
+# 
+# Install Homebrew in a Terminal via
+# $ /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+# 
+# Install the following via Homebrew
+# $ brew install cmake
+# $ brew install openmpi --c++11 --with-mpi-thread-multiple
```
Comment (tjhei) [inline deal.II-toolchain/platforms/supported/sierra.platform]: Do you need those two flags?

File: deal.II-toolchain/platforms/supported/sierra.platform [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+# 
+# (Optional) Install the following via Homebrew
+# $ brew install wget
+# $ brew install gnuplot
+# 
+# $ brew install bash
```
Comment (tjhei) [inline deal.II-toolchain/platforms/supported/sierra.platform]: Does candi.sh run for you without installing bash4 using mac osx?

**Comment by tjhei** [comment]
Looks good in general. I have a few questions inline above. If you want I can run on mac osx later (in a few days)

File: deal.II-toolchain/platforms/supported/sierra.platform (hunk 7 above)
Comment (koecher) [inline deal.II-toolchain/platforms/supported/sierra.platform]: Yes, but it shows an error for parsing if statements with double brackets [[ ]]. We use them to switch on / off packages.

File: deal.II-toolchain/platforms/supported/sierra.platform (hunk 6 above)
Comment (koecher) [inline deal.II-toolchain/platforms/supported/sierra.platform]: I'm not an expert in brew, but from their name I would like to have c++11 support (needed in my software) and also threads within a mpi process.
Maybe c++11 support is handled by the underlying apple clang anyway, I don't know.

**Comment by koecher** [comment]
Now I'm looking for the oce problem and will make some tests on my linux workstations / clusters. Then it should be fine.

**Comment by koecher** [comment]
@tjhei for opencascade the release tarball should be updated to v0.18

**Comment by koecher** [comment]
My test runs on centos 7 and fedora 25 were successful, from my point of view this PR is ready to merge.
Note: please merge only if @tjhei recommends this.

**Comment by koecher** [comment]
@tjhei Have you done a test run?

**Comment by tjhei** [comment]
Sorry for the delay, Uwe. I am testing ubuntu and my mac right now.

**Comment by tjhei** [comment]
ubuntu works, osx fails for me in petsc (something about fortran compiler is unable to compile a program). I am not sure what is going on.

**Comment by koecher** [comment]
Are you using the compilers as written in the readme output, or are you using "your own" ones?
I also had a petsc problem, but using version 3.7.6 works for me, I've added those related commits on May 6th.

**Comment by tjhei** [comment]
I am not 100% sure (I played around with gcc before, but I thought I was using clang this time). I can not check because my macbook that sits in my office in the US decided to go offline and I am in Europe for the next couple of weeks.
Anyways, I think this is better than before, so I am okay with merging as is.

**Comment by koecher** [comment]
Alright, then I merge. Thanks for your statement.

**Pull request merged**